### DISCUSSION TITLE
How do i use this crate to find eigenvalues and eigenvectors of sparse matrix

### DISCUSSION THREAD

**Discussion by iverks** [vbarrielle/sprs#325] [Q&A] [answered]
How do i find eigenvalues and eigenvectors of a sparse matrix? I have little knowledge of the history of matrices, so i find the names ldl and csc and so on to be kind of confusing. I figured out how to create a TriMat, and how to convert said TriMat into a Ldl, but cannot figure out how to get the eigenvalues and vectors from it. Would be great if there was an example on how to do this in the docs.

Alternatively can someone point me in the right direction? The matrix needs to be sparse, but is not tridiagonal

**Comment by mulimoen** [comment, marked as answer]
There is no one-size-fits-all when it comes to eigendecomposition, and we have not implemented any methods for finding the eigenvalues/vectors. You likely want to use lapack or the like to take advantage of a known good library

**Reply by mulimoen** [reply]
Finding eigenvectors is also a very expensive problem, consider using https://en.m.wikipedia.org/wiki/ARPACK if your matrices are very large

**Reply by iverks** [reply]
Thanks for the answers, i ended up using python + scipy to solve this since they have an easy to use interface for solving sparse matrices. Hope you guys figure out a nice implementation/abstraction later since i loved using rust + ndarray + nalgebra for the rest of my project, and would love to recommend it to the students of the same numerics course in the future.
